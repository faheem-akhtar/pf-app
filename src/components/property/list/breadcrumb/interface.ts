export interface BreadcrumbInterface {
  /**
   * Name
   * @example Apartments
   */
  name: string;

  /**
   * Routing parameters
   * @example /en/search?t=1&c=1
   */
  link: string;

  /**
   * Total count for the link
   * @example 4881
   */
  count: number;
}
