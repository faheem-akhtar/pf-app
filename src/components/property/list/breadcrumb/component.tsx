import { useRouter } from 'next/router';
import { FunctionComponent, useContext } from 'react';

import { FiltersContext } from 'components/filters/context';

import { PropertyListBreadcrumbComponentPropsInterface } from './component-props.interface';
import { breadcrumbMapQueryToFilters } from './map-query-to-filters';
import { PropertyListBreadcrumbTemplate } from './template';
import { PropertyListBreadcrumbTemplatePropsInterface } from './template-props.interface';

export const PropertyListBreadcrumbComponent: FunctionComponent<PropertyListBreadcrumbComponentPropsInterface> = (
  props
) => {
  const locale = useRouter().locale as string;
  const filtersCtx = useContext(FiltersContext);

  const onClick: PropertyListBreadcrumbTemplatePropsInterface['onClick'] = (event) => {
    event.preventDefault();
    filtersCtx.change((filtersValue) => ({
      ...filtersValue,
      ...breadcrumbMapQueryToFilters(locale, props.link),
    }));
  };

  return <PropertyListBreadcrumbTemplate {...props} onClick={onClick} />;
};
