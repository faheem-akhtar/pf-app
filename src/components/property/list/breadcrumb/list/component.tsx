import { FunctionComponent } from 'react';

import { IconThinChevronDownTemplate } from 'components/icon';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonIconPositionEnum } from 'library/button/icon-position.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';

import { PropertyListBreadcrumbComponent } from '../component';
import styles from '../property-list-breadcrumb.module.scss';
import { PropertyListBreadcrumbListComponentPropsInterface } from './component-props.interface';
import { propertyListBreadcrumbListItemsVisibleCount } from './items-visible-count';

export const PropertyListBreadcrumbListComponent: FunctionComponent<PropertyListBreadcrumbListComponentPropsInterface> =
  ({ breadcrumbs, onClickShowMore, expanded, t }) => (
    <section role='section' className={styles.breadcrumb}>
      <ul role='list' className={styles.breadcrumb__list}>
        {breadcrumbs.slice(0, propertyListBreadcrumbListItemsVisibleCount).map((breadcrumb) => (
          <PropertyListBreadcrumbComponent key={breadcrumb.name} {...breadcrumb} />
        ))}

        {breadcrumbs.length > propertyListBreadcrumbListItemsVisibleCount && (
          <li role='listitem'>
            <ButtonTemplate
              className={styles.breadcrumb__button}
              type='button'
              componentType={ButtonComponentTypeEnum.secondaryBlue}
              size={ButtonSizeEnum.small}
              onClick={onClickShowMore}
              icon={{
                component: IconThinChevronDownTemplate,
                position: ButtonIconPositionEnum.right,
                className: domClassMerge(styles.breadcrumb__icon, {
                  [styles['breadcrumb__icon--expanded']]: expanded,
                }),
              }}
            >
              {t(expanded ? 'show-less' : 'show-more')}
            </ButtonTemplate>
          </li>
        )}
      </ul>

      <ul
        className={domClassMerge(styles.breadcrumb__list, styles['breadcrumb__list--more'], {
          [styles['breadcrumb__list--expanded']]: expanded,
        })}
      >
        {breadcrumbs.slice(propertyListBreadcrumbListItemsVisibleCount, breadcrumbs.length).map((breadcrumb) => (
          <PropertyListBreadcrumbComponent
            key={breadcrumb.name}
            class={styles['breadcrumb__item--more']}
            {...breadcrumb}
          />
        ))}
      </ul>
    </section>
  );
