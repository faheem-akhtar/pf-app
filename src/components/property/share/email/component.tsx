import { Fragment, useState } from 'react';

import { IconThinCheckmarkCircleTemplate } from 'components/icon/thin/checkmark-circle-template';
import { propertySerpObfuscatedGetId } from 'components/property/serp/obfuscated/get/id';
import { propertySerpObfuscatedGetImgUrl } from 'components/property/serp/obfuscated/get/img-url';
import { propertySerpObfuscatedGetName } from 'components/property/serp/obfuscated/get/name';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';

import { propertyShareTracker } from '../tracker';
import { PropertyShareEmailComponentPropsInterface } from './component-props.interface';
import { PropertyShareEmailFormComponent } from './form/component';
import styles from './property-share-email.module.scss';

export const PropertyShareEmailComponent = ({
  property,
  t,
  onClickClose,
}: PropertyShareEmailComponentPropsInterface): JSX.Element => {
  const [isFormSent, setIsFormSent] = useState(false);
  const propertyImageUrl = propertySerpObfuscatedGetImgUrl(property);
  const propertyName = propertySerpObfuscatedGetName(property);

  if (!isFormSent) {
    return (
      <Fragment>
        <div className={styles.info}>
          <div className={styles.imageContainer}>
            <picture className={styles.image}>
              <img src={propertyImageUrl} className={styles.image} alt={propertyName} />
            </picture>
          </div>
          <div className={styles.title}>{propertyName}</div>
        </div>
        <PropertyShareEmailFormComponent
          t={t}
          propertyId={propertySerpObfuscatedGetId(property)}
          onFormSubmitted={(): void => {
            setIsFormSent(true);
            propertyShareTracker.onSuccessSocialShare('Email');
          }}
        />
      </Fragment>
    );
  }

  return (
    <div className={styles.success}>
      <div className={styles.successContent}>
        <IconThinCheckmarkCircleTemplate class={styles.successIcon} />
        <div className={styles.successContent}>
          <p>{t('thank-you')}</p>
          <p>{t('your-message-is-sent')}</p>
        </div>
      </div>
      <ButtonTemplate
        type='button'
        size={ButtonSizeEnum.regular}
        componentType={ButtonComponentTypeEnum.primary}
        onClick={onClickClose}
        fullWidth
      >
        {t('close')}
      </ButtonTemplate>
    </div>
  );
};
