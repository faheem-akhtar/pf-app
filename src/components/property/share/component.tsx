import { useRef, useState } from 'react';

import { ModalComponent } from 'components/modal/component';
import { propertySerpObfuscatedGetUrl } from 'components/property/serp/obfuscated/get/url';
import { domClassMerge } from 'helpers/dom/class-merge';
import { functionNoop } from 'helpers/function/noop';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';

import { PropertyShareComponentPropsInterface } from './component-props.interface';
import { PropertyShareEmailComponent } from './email/component';
import styles from './property-share.module.scss';
import { PropertyShareSocialComponent } from './social/component';

export const PropertyShareComponent = ({ t, openRef, property }: PropertyShareComponentPropsInterface): JSX.Element => {
  const [shareViaEmail, setShareViaEmail] = useState(false);
  const closeRef = useRef<() => void>(functionNoop);

  const propertyUrl = propertySerpObfuscatedGetUrl(property);

  const closeModal = (): void => {
    setShareViaEmail(false);
    closeRef.current();
  };

  return (
    <ModalComponent
      openRef={openRef}
      closeRef={closeRef}
      overlay
      containerClassName={styles.overlay}
      onOverlayClick={closeModal}
    >
      <div
        data-testid='property-share'
        className={styles.container}
        onClick={(e): void => {
          e.stopPropagation();
        }}
      >
        <div
          className={domClassMerge(styles.title, {
            [styles['title--right']]: !shareViaEmail,
          })}
        >
          {shareViaEmail && <h1 className={styles.emailTitle}>{t('cta-share')}</h1>}
          <ButtonTemplate
            className={styles.close}
            onClick={closeModal}
            type='button'
            componentType={ButtonComponentTypeEnum.secondaryBlue}
            size={ButtonSizeEnum.small}
          >
            {t('close')}
          </ButtonTemplate>
        </div>
        {!shareViaEmail ? (
          <PropertyShareSocialComponent
            propertyUrl={propertyUrl}
            t={t}
            onClickEmail={(): void => {
              setShareViaEmail(true);
            }}
          />
        ) : (
          <PropertyShareEmailComponent onClickClose={closeModal} property={property} t={t} />
        )}
      </div>
    </ModalComponent>
  );
};
