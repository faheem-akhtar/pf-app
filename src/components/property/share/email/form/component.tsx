import { FormEvent, useState } from 'react';

import { apiShareFetcher } from 'api/share/fetcher';
import { formMakeValidator } from 'components/form/make-validator';

import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { configSiteNameValue } from 'config/site-name/value';
import { ErrorMessageComponent } from 'components/error-message/component';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import { GoogleRecaptchaService } from 'services/google-recaptcha/service';
import { InputBaseComponent } from 'library/input/base/component';
import { PropertyShareEmailFieldsEnum } from '../fields.enum';
import { PropertyShareEmailFormComponentPropsInterface } from './component-props.interface';
import { ReCaptchaComponent } from 'components/re-captcha/component';

import styles from './property-share-email-form.module.scss';

const captchaService = new GoogleRecaptchaService();

export const PropertyShareEmailFormComponent = ({
  propertyId,
  onFormSubmitted,
  t,
}: PropertyShareEmailFormComponentPropsInterface): JSX.Element => {
  const [email, setEmail] = useState<string>('');
  const [friendEmail, setFriendEmail] = useState<string>('');
  const [message, setMessage] = useState<string>(
    t('social-share/default-message').replace('{sitename}', t(configSiteNameValue))
  );
  const [requestErrorMessage, setRequestErrorMessage] = useState('');
  const [errors, setErrors] = useState<Partial<Record<PropertyShareEmailFieldsEnum, string>>>({});
  const [isLoading, setIsLoading] = useState(false);
  const [validators] = useState(() => ({
    [PropertyShareEmailFieldsEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
    [PropertyShareEmailFieldsEnum.friendsEmail]: [
      validationRequired(t('social-share/empty-friends-mail')),
      validationEmail(t('validation/not-valid-email')),
    ],
    [PropertyShareEmailFieldsEnum.message]: [validationRequired(t('social-share/not-empty-message'))],
  }));

  const validate = formMakeValidator(errors, setErrors, validators);

  const onSubmitForm = async (e: FormEvent<HTMLElement>): Promise<void> => {
    e.preventDefault();
    setRequestErrorMessage('');
    if (
      !validate({
        [PropertyShareEmailFieldsEnum.email]: email,
        [PropertyShareEmailFieldsEnum.friendsEmail]: friendEmail,
        [PropertyShareEmailFieldsEnum.message]: message,
      })
    ) {
      setIsLoading(true);
      const captcha_token = await captchaService.execute();

      const shareProperty = await apiShareFetcher(propertyId, {
        email,
        message,
        friend_email: friendEmail,
        captcha_token,
      });
      if (shareProperty.ok !== null) {
        setIsLoading(false);
      }

      if (shareProperty.ok) {
        onFormSubmitted();
      } else {
        setRequestErrorMessage(t('something-wrong-try-again'));
        setIsLoading(false);
        captchaService.reset();
      }
    }
  };

  return (
    <form onSubmit={onSubmitForm}>
      <div className={styles.field}>
        <InputBaseComponent
          value={email}
          type='email'
          inputClassName={styles.input}
          error={!!errors.email}
          errorText={errors.email}
          onChange={(value): void => {
            validate({ [PropertyShareEmailFieldsEnum.email]: value });
            setEmail(value);
          }}
          placeholder={t('your-email')}
        />
      </div>
      <div className={styles.field}>
        <InputBaseComponent
          type='email'
          value={friendEmail}
          inputClassName={styles.input}
          error={!!errors.friendsEmail}
          errorText={errors.friendsEmail}
          onChange={(value): void => {
            validate({ [PropertyShareEmailFieldsEnum.friendsEmail]: value });
            setFriendEmail(value);
          }}
          placeholder={t('friends-email')}
        />
      </div>
      <div className={styles.field}>
        <InputBaseComponent
          value={message}
          inputClassName={styles.textarea}
          error={!!errors.message}
          textarea
          errorText={errors.message}
          onChange={(value): void => {
            validate({ [PropertyShareEmailFieldsEnum.message]: value });
            setMessage(value);
          }}
        />
      </div>
      <div className={styles.field}>
        {requestErrorMessage && <ErrorMessageComponent message={requestErrorMessage} />}
        <ButtonTemplate
          type='submit'
          loading={isLoading}
          size={ButtonSizeEnum.regular}
          componentType={ButtonComponentTypeEnum.primary}
          fullWidth
        >
          {t('send')}
        </ButtonTemplate>
      </div>
      <ReCaptchaComponent />
    </form>
  );
};
