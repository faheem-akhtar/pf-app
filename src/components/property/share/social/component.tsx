import { Fragment } from 'react';

import { IconShareEmailTemplate } from 'components/icon/share/email-template';
import { IconShareFacebookTemplate } from 'components/icon/share/facebook-template';
import { IconShareTwitterTemplate } from 'components/icon/share/twitter-template';
import { IconShareWhatsappTemplate } from 'components/icon/share/whatsapp-template';
import { domClassMerge } from 'helpers/dom/class-merge';

import styles from '../property-share.module.scss';
import { propertyShareTracker } from '../tracker';
import { PropertyShareSocialComponentPropsInterface } from './component-props.interface';

const socialGetShareLinkUtmPlatform = (url: string, platform: string): string =>
  `${url}?utm_source=${platform}&utm_medium=social&utm_campaign=share_property`;

export const PropertyShareSocialComponent = ({
  propertyUrl,
  t,
  onClickEmail,
}: PropertyShareSocialComponentPropsInterface): JSX.Element => {
  return (
    <Fragment>
      <div className={styles.header}>{t('social-share/title')}</div>
      <div className={styles.content}>
        <a
          className={styles.link}
          href={`https://www.facebook.com/sharer/sharer.php?u=${socialGetShareLinkUtmPlatform(
            propertyUrl,
            'facebook'
          )}`}
          target='_blank'
          rel='noreferrer'
          onClick={(): void => propertyShareTracker.onClickSocialShare('Facebook')}
        >
          <IconShareFacebookTemplate class={domClassMerge(styles.icon, styles.facebook)} />
        </a>
        <a
          className={styles.link}
          href={`whatsapp://send?text=${socialGetShareLinkUtmPlatform(propertyUrl, 'whatsapp')}`}
          data-action='share/whatsapp/share'
          target='_blank'
          rel='noreferrer'
          onClick={(): void => propertyShareTracker.onClickSocialShare('Whatsapp')}
        >
          <IconShareWhatsappTemplate class={domClassMerge(styles.icon, styles.whatsapp)} />
        </a>
        <div
          className={styles.link}
          onClick={(): void => {
            propertyShareTracker.onClickSocialShare('Email');
            onClickEmail();
          }}
        >
          <IconShareEmailTemplate class={domClassMerge(styles.icon, styles.email)} />
        </div>
        <a
          className={styles.link}
          href={`https://twitter.com/intent/tweet?url=${socialGetShareLinkUtmPlatform(propertyUrl, 'twitter')}`}
          target='_blank'
          rel='noreferrer'
          onClick={(): void => propertyShareTracker.onClickSocialShare('Twitter')}
        >
          <IconShareTwitterTemplate class={domClassMerge(styles.icon, styles.twitter)} />
        </a>
      </div>
    </Fragment>
  );
};
