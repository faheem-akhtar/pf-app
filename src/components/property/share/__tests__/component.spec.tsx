import { act, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { mockModalEnv } from 'mocks/modal-env/mock';
import { mockWindowFetch } from 'mocks/window/fetch.mock';
import { googleRecaptchaStub } from 'stubs/google/recaptcha.stub';
import { propertyStub } from 'stubs/property/stub';

import { propertySerpObfuscatedGetId } from 'components/property/serp/obfuscated/get/id';
import { propertySerpObfuscatedGetName } from 'components/property/serp/obfuscated/get/name';
import { PropertySerpObfuscatedType } from 'components/property/serp/obfuscated/type';
import { functionSelf } from 'helpers/function/self';
import { GoogleRecaptcha } from 'services/google/recaptcha';
import * as GoogleRecaptchaServiceModule from 'services/google/recaptcha.service';

import { PropertyShareComponent } from '../component';
import { PropertyShareComponentPropsInterface } from '../component-props.interface';

describe('PropertyShareComponent', () => {
  let props: PropertyShareComponentPropsInterface;
  const openRef = { current: jest.fn() };
  const closeRef = { current: jest.fn() };
  const property: PropertySerpObfuscatedType = propertyStub();
  const propertyName = propertySerpObfuscatedGetName(property);
  const propertyId = propertySerpObfuscatedGetId(property);

  beforeEach(() => {
    mockModalEnv();

    window.dataLayer = [];

    props = {
      openRef,
      property,
      t: functionSelf,
    };
  });

  it('renders without throwing any errors', () => {
    render(<PropertyShareComponent {...props} />);
    act(openRef.current);

    expect(screen.getByTestId('property-share')).toMatchSnapshot();
  });

  it('should work the email form functionality properly', () => {
    render(<PropertyShareComponent {...props} />);
    act(openRef.current);

    userEvent.click(screen.getByText(/email/i));

    expect(screen.getByText(propertyName)).toBeInTheDocument();
    expect(screen.queryByRole('heading', { level: 1, name: 'social-share/title' })).not.toBeInTheDocument();

    userEvent.click(screen.getByRole('button', { name: 'close' }));
    act(closeRef.current);

    expect(screen.queryByTestId('property-share')).not.toBeInTheDocument();
  });

  describe('Email Form', () => {
    let fetchMock: jest.Mock;
    let googleRecaptchaMock: GoogleRecaptcha;

    beforeEach(() => {
      fetchMock = mockWindowFetch();

      googleRecaptchaMock = googleRecaptchaStub();
      jest.spyOn(GoogleRecaptchaServiceModule, 'GoogleRecaptchaService').mockReturnValue(googleRecaptchaMock);

      render(<PropertyShareComponent {...props} />);
      act(openRef.current);

      userEvent.click(screen.getByText(/email/i));
    });

    it('should not submit the form and display the empty email error', () => {
      userEvent.type(screen.getByLabelText('friends-email'), 'friend@gmail.com');
      userEvent.click(screen.getByRole('button', { name: 'send' }));

      expect(screen.getByText('validation/empty-email')).toBeInTheDocument();
    });

    it('should not submit the form and display the not valid email error', () => {
      userEvent.type(screen.getByLabelText('email'), 'e');
      userEvent.type(screen.getByLabelText('friends-email'), 'friend@g.com');
      userEvent.click(screen.getByRole('button', { name: 'send' }));

      expect(screen.getByText('validation/not-valid-email')).toBeInTheDocument();
    });

    it('should not submit the form and display the empty friends email error', () => {
      userEvent.type(screen.getByLabelText('email'), 'email@g.com');
      userEvent.click(screen.getByRole('button', { name: 'send' }));

      expect(screen.getByText('social-share/empty-friends-mail')).toBeInTheDocument();
    });

    it('should not submit the form and display the not valid email error for friends email', () => {
      userEvent.type(screen.getByLabelText('email'), 'email@g.com');
      userEvent.type(screen.getByLabelText('friends-email'), 'f');
      userEvent.click(screen.getByRole('button', { name: 'send' }));

      expect(screen.getByText('validation/not-valid-email')).toBeInTheDocument();
    });

    it('should not submit the form and display the not empty message error for message', () => {
      userEvent.type(screen.getByLabelText('email'), 'email@g.com');
      userEvent.type(screen.getByLabelText('friends-email'), 'friends@g.com');
      userEvent.clear(screen.getByLabelText('message'));
      userEvent.click(screen.getByRole('button', { name: 'send' }));

      expect(screen.getByText('social-share/not-empty-message')).toBeInTheDocument();
    });

    it('should display general error message if the request failed', async () => {
      fetchMock = mockWindowFetch({ ok: false, status: 500 });

      userEvent.type(screen.getByLabelText('email'), 'email@g.com');
      userEvent.type(screen.getByLabelText('friends-email'), 'friends@g.com');
      userEvent.click(screen.getByRole('button', { name: 'send' }));

      await screen.findByText('something-wrong-try-again');

      expect(screen.getByText('something-wrong-try-again'));

      expect(googleRecaptchaMock.execute).toHaveBeenCalledTimes(1);
      expect(googleRecaptchaMock.reset).toHaveBeenCalledTimes(1);
      expect(fetchMock).toHaveBeenCalledTimes(1);
    });

    it('should submit the form and display success', async () => {
      userEvent.type(screen.getByLabelText('email'), 'email@g.com');
      userEvent.type(screen.getByLabelText('friends-email'), 'friends@g.com');
      userEvent.click(screen.getByRole('button', { name: 'send' }));

      await waitFor(() =>
        expect(fetchMock).toHaveBeenCalledWith(
          expect.stringContaining(`default-origin/en/api/property/${propertyId}/share`),
          {
            body: JSON.stringify({
              data: {
                type: 'property_share',
                attributes: {
                  email: 'email@g.com',
                  message: 'social-share/default-message',
                  friend_email: 'friends@g.com',
                  captcha_token: 'token',
                },
              },
            }),
            headers: {
              locale: 'en',
              'content-type': 'application/vnd.api+json',
            },
            method: 'POST',
          }
        )
      );

      expect(googleRecaptchaMock.execute).toHaveBeenCalledTimes(1);
      expect(googleRecaptchaMock.reset).not.toHaveBeenCalled();
      expect(fetchMock).toHaveBeenCalledTimes(1);

      expect(screen.getByText('thank-you')).toBeInTheDocument();
      expect(screen.getByText('your-message-is-sent')).toBeInTheDocument();

      expect(window.dataLayer).toEqual(
        expect.arrayContaining([
          {
            event: 'User Interaction',
            eventAction: 'Finish:Email Share',
            eventLabel: 'Social Sharing',
          },
        ])
      );
    });
  });

  describe('GA events', () => {
    beforeEach(() => {
      render(<PropertyShareComponent {...props} />);
      act(openRef.current);
    });

    it('should send an event when facebook is clicked', () => {
      userEvent.click(screen.getByRole('link', { name: /facebook/i }));

      expect(window.dataLayer).toEqual(
        expect.arrayContaining([
          {
            event: 'User Interaction',
            eventAction: 'Click:Facebook Share',
            eventLabel: 'Social Sharing',
          },
        ])
      );
    });

    it('should send an event when whatsapp is clicked', () => {
      userEvent.click(screen.getByRole('link', { name: /whatsapp/i }));

      expect(window.dataLayer).toEqual(
        expect.arrayContaining([
          {
            event: 'User Interaction',
            eventAction: 'Click:Whatsapp Share',
            eventLabel: 'Social Sharing',
          },
        ])
      );
    });

    it('should send an event when twitter is clicked', () => {
      userEvent.click(screen.getByRole('link', { name: /twitter/i }));

      expect(window.dataLayer).toEqual(
        expect.arrayContaining([
          {
            event: 'User Interaction',
            eventAction: 'Click:Twitter Share',
            eventLabel: 'Social Sharing',
          },
        ])
      );
    });
  });
});
