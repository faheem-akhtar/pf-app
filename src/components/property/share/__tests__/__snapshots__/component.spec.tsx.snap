// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyShareComponent renders without throwing any errors 1`] = `
<div
  class="container"
  data-testid="property-share"
>
  <div
    class="title title--right"
  >
    <button
      class="button secondaryBlue small close"
    >
      close
    </button>
  </div>
  <h1
    class="header"
  >
    social-share/title
  </h1>
  <div
    class="content"
  >
    <a
      class="link"
      href="https://www.facebook.com/sharer/sharer.php?u=url?utm_source=facebook&utm_medium=social&utm_campaign=share_property"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        class="icon facebook"
        viewBox="0 0 42 42"
      >
        <title>
          Facebook
        </title>
        <g
          stroke="none"
          stroke-width="1"
        >
          <g
            transform="translate(-1.000000, -1.000000)"
          >
            <path
              d="M22.0000372,1 C33.5979907,1 43.0000372,10.4020464 43.0000372,22 C43.0000372,33.5983279 33.5979907,43 22.0000372,43 C10.4020836,43 1.00003717,33.5983279 1.00003717,22 C1.00003717,10.4020464 10.4020836,1 22.0000372,1 Z M27.0324839,22.1206851 L27.4622504,18.2010021 L23.7860445,18.2010021 L23.7860445,15.8919934 C23.7860445,15.0259248 24.3958263,14.8225738 24.8294849,14.8225738 L27.472954,14.8225738 L27.472954,11.0143076 L23.8295076,11 C19.7854865,11 18.8665972,13.8429574 18.8665972,15.6582005 L18.8665972,18.197349 L16.527046,18.197349 L16.527046,22.1206851 L18.8665972,22.1206851 L18.8665972,33.2842931 L23.7857201,33.2842931 L23.7857201,22.1206851 L27.0324839,22.1206851 Z"
            />
          </g>
        </g>
      </svg>
    </a>
    <a
      class="link"
      data-action="share/whatsapp/share"
      href="whatsapp://send?text=url?utm_source=whatsapp&utm_medium=social&utm_campaign=share_property"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        class="icon whatsapp"
        viewBox="0 0 42 42"
      >
        <title>
          Whatsapp
        </title>
        <g
          stroke="none"
          stroke-width="1"
        >
          <g
            id="DESKTOP---Social-Share-(Hover-State)-Copy-5"
            transform="translate(-1.000000, -1.000000)"
          >
            <path
              d="M22.0000372,1 C33.5979907,1 43.0000372,10.4020464 43.0000372,22 C43.0000372,33.5983279 33.5979907,43 22.0000372,43 C10.4020836,43 1.00003717,33.5983279 1.00003717,22 C1.00003717,10.4020464 10.4020836,1 22.0000372,1 Z M10,34 L16.305,32.346 C18.042,33.294 20.003,33.793 21.993,33.794 C28.548,33.794 33.883,28.459 33.886,21.902 C33.887,18.724 32.651,15.736 30.406,13.488 C28.16,11.24 25.174,10.001 21.993,10 C15.438,10 10.103,15.335 10.1,21.891 C10.099,23.988 10.646,26.033 11.687,27.837 L10,34 Z M16.597,30.193 L12.855,31.174 L13.854,27.526 C12.759,25.783 12.107,24.117 12.108,21.892 C12.11,16.442 16.545,12.008 21.997,12.008 C27.463,12.01 31.88,16.438 31.878,21.9 C31.875,27.351 27.437,31.785 21.989,31.785 C19.873,31.784 18.273,31.188 16.597,30.193 Z M27.984,24.729 C27.91,24.605 27.712,24.531 27.414,24.382 C27.117,24.233 25.656,23.514 25.383,23.416 C25.111,23.316 24.913,23.266 24.714,23.564 C24.516,23.861 23.946,24.531 23.773,24.729 C23.6,24.927 23.426,24.952 23.129,24.803 C22.832,24.654 21.874,24.341 20.739,23.328 C19.856,22.54 19.259,21.567 19.086,21.269 C18.913,20.972 19.068,20.811 19.216,20.663 C19.35,20.53 19.513,20.316 19.662,20.142 C19.813,19.97 19.862,19.846 19.962,19.647 C20.061,19.449 20.012,19.275 19.937,19.126 C19.862,18.978 19.268,17.515 19.021,16.92 C18.779,16.341 18.534,16.419 18.352,16.41 L17.782,16.4 C17.584,16.4 17.262,16.474 16.99,16.772 C16.718,17.07 15.95,17.788 15.95,19.251 C15.95,20.714 17.015,22.127 17.163,22.325 C17.312,22.523 19.258,25.525 22.239,26.812 C22.948,27.118 23.502,27.301 23.933,27.438 C24.645,27.665 25.293,27.632 25.805,27.556 C26.376,27.471 27.563,26.837 27.811,26.143 C28.059,25.448 28.059,24.853 27.984,24.729 Z"
            />
          </g>
        </g>
      </svg>
    </a>
    <div
      class="link"
    >
      <svg
        class="icon email"
        viewBox="0 0 42 42"
      >
        <title>
          Email
        </title>
        <g
          stroke="none"
          stroke-width="1"
        >
          <g
            transform="translate(-1.000000, -1.000000)"
          >
            <path
              d="M22,1 C33.5979536,1 43,10.4020464 43,22 C43,33.5983279 33.5979536,43 22,43 C10.4020464,43 1,33.5983279 1,22 C1,10.4020464 10.4020464,1 22,1 Z M10.5,15 L10.5,15.7796176 L10.5,16.9007982 L10.5,28.2159272 L10.5,28.6215333 L10.5,30.1167255 L33.5,30.1167255 L33.5,28.6215333 L33.5,28.2159272 L33.5,16.902506 L33.5,15.7796176 L33.5,15 L10.5,15 Z M13.7499722,16.902506 L30.3627436,16.902506 L22.0542231,22.7884908 L13.7499722,16.902506 Z M12.4161686,18.3029144 L17.8931316,22.1856321 L12.4161686,27.0443661 L12.4161686,18.3029144 Z M26.1768888,22.2146649 L31.5838314,18.3840356 L31.5838314,26.943605 L26.1768888,22.2146649 Z M19.4993317,23.3238908 L22.0439762,25.1290514 L22.0499536,25.1384444 L22.0542231,25.1341749 L22.0576388,25.1384444 L22.0627622,25.1290514 L24.5689809,23.3546315 L30.1279191,28.2167811 L13.9830889,28.2167811 L19.4993317,23.3238908 Z"
            />
          </g>
        </g>
      </svg>
    </div>
    <a
      class="link"
      href="https://twitter.com/intent/tweet?url=url?utm_source=twitter&utm_medium=social&utm_campaign=share_property"
      rel="noreferrer"
      target="_blank"
    >
      <svg
        class="icon twitter"
        viewBox="0 0 42 42"
      >
        <title>
          Twitter
        </title>
        <g
          stroke="none"
          stroke-width="1"
        >
          <g
            transform="translate(-1.000000, -1.000000)"
          >
            <path
              d="M22.0000372,1 C33.5979907,1 43.0000372,10.4020464 43.0000372,22 C43.0000372,33.5983279 33.5979907,43 22.0000372,43 C10.4020836,43 1.00003717,33.5983279 1.00003717,22 C1.00003717,10.4020464 10.4020836,1 22.0000372,1 Z M34.146177,14.4662035 C33.2526549,14.8624159 32.2914867,15.1303982 31.2834867,15.2500796 C32.3126726,14.6334602 33.1024956,13.6577965 33.475292,12.4933186 C32.5122655,13.0645929 31.4451681,13.4790177 30.3104248,13.7023982 C29.401292,12.7341681 28.1059823,12.1290708 26.6716637,12.1290708 C23.9197345,12.1290708 21.6877876,14.3610177 21.6877876,17.1129469 C21.6877876,17.5035841 21.7320177,17.8838142 21.8175044,18.2488053 C17.6754867,18.0410354 14.0029027,16.057 11.5446018,13.0408053 C11.1156814,13.7767345 10.8696283,14.6334602 10.8696283,15.5466814 C10.8696283,17.2753717 11.7501416,18.8014956 13.086708,19.6950177 C12.2701239,19.6693717 11.501115,19.4452478 10.8294867,19.0709646 C10.829115,19.0921504 10.829115,19.1133363 10.829115,19.1341504 C10.829115,21.5493363 12.5477699,23.5634779 14.827292,24.020646 C14.4095221,24.1354956 13.9683363,24.195708 13.5145133,24.195708 C13.1926372,24.195708 12.8807965,24.1648584 12.5767611,24.1068761 C13.2108496,26.086823 15.0510442,27.5278319 17.2324425,27.5679735 C15.5264248,28.9052832 13.3777345,29.7014248 11.0420885,29.7014248 C10.6406726,29.7014248 10.2429735,29.6780088 9.85382301,29.6315487 C12.0582655,31.0461681 14.6786195,31.8709292 17.4929912,31.8709292 C26.6601416,31.8709292 31.6733805,24.2771062 31.6733805,17.6901681 C31.6733805,17.4742212 31.6685487,17.2590177 31.658885,17.0456726 C32.6334336,16.3431947 33.4782655,15.4656549 34.146177,14.4662035 Z"
            />
          </g>
        </g>
      </svg>
    </a>
  </div>
</div>
`;
